MIP/mip-103/README.md

```diff
@@ -0,0 +1,257 @@
+      commitments = self.commitments[height]
+      stake_weights = self.stake_weights[epoch]
+
+      # Check if the sum of the stake weights voting on a given commitment is greater than the threshold
+      if sum(stake_weights[participant] for participant in commitments if commitments[participant] == commitment) > threshold:
+          # Compute the slashing and rewarding amounts
+          slashing_amounts = S(commitments, stake_weights, commitment, height, epoch)
+          rewarding_amounts = R(commitments, stake_weights, commitment, height, epoch)
+
+          # Return the proof of consensus and the slashing and rewarding amounts
+          return proof, slashing_amounts, rewarding_amounts
+      else:
+          return None, None, None
+
+  ```
+
+  ##### Human Language Description of Heuristic
+
+  This MIP does prescribe a specific heuristic for deciding when to run $COMPUTE\\_CONSENSUS$. In general, however, the **Validator** would save compute by only running $COMPUTE\\_CONSENSUS$ when it has received a sufficient number of commitments and stake weights, i.e., by precomputing the $COMPUTE\\_CONSENSUS$ procedure outside of the ZK execution environment.
+
+#### $POST\\_PROOF$
+
+- Proof posting is intended to be a simple procedure which broadcasts a $COMMITMENT\\_PROOF$ message to A-Participants. The message SHALL include the commitment, height, Attesters, proof, slashing amounts, rewarding amounts, and signature. It does not require any additional computation.
+
+  ##### Human Language Description of Heuristic
+
+  This MIP does prescribe a specific heuristic for deciding when to run $POST\\_PROOF$. In general, however, the **Validator** has to balance on-chain costs with the expected value of the reward.
+
+  To avoid disincentivizing honest posting behavior due to reward uncertainty, the on-chain reward mechanism should accept more than simply the initial commit. This phenomenon is addresses more completely in [Phenomena](#phenomena).
+
+### Phenomena
+
+- #### Forking and Asynchrony
+
+  In general, forks are easier to represent off-chain than on-chain. We, however, note several distinct simplifications which arise from the Baker Confirmations protocol.
+
+  ##### Fork Representation
+
+  Baker Confirmations for a Baker Coin Protocol remove the need to represent forks on-chain. The on-chain contract must simply verify the Zero Knowledge proof of consensus and dispatch the appropriate onchain events.
+
+  ##### Fork Stake
+
+  The primary problem with stake in an FFS confirmation device, outlined in [MD-3](https://github.com/movementlabsxyz/MIP/pull/3/files) is that--because said device is not inherently designed within the reorganization logic of the confirming ledger itself--rewards, once issued on a given fork, cannot be revoked. In contrast, in more conventional DLT, the rewards exist on the fork of the ledger itself and once that fork is orphaned, the rewards are revoked.
```

Comment:
Are you refering to this file? "MCR: Asynchronous Upgrades, Fork Creation, and Fork Stake Problems"
it is no longer part of PR3 but part of #101

```suggestion
  The primary problem with stake in an FFS confirmation device, outlined in [MD-3](https://github.com/movementlabsxyz/MIP/blob/md/mcr-asynchrony-part-2/MD/md-3/asychronous-upgrades-problem.md) is that--because said device is not inherently designed within the reorganization logic of the confirming ledger itself--rewards, once issued on a given fork, cannot be revoked. In contrast, in more conventional DLT, the rewards exist on the fork of the ledger itself and once that fork is orphaned, the rewards are revoked.
```